### PR TITLE
WT-7743 Fix integer overflow within wt2999_join_extractor csuite test

### PR DESCRIPTION
**Pull request by AlexeyAnisimovAU**

### CODE REVIEW

**Review by luke-pearson — APPROVED** [review]
LGTM. Maybe add a comment too so its a bit clearer as to why its not just a regular int.

File: test/csuite/wt2999_join_extractor/main.c [hunk 1]
```diff
@@ -86,8 +86,9 @@ main(int argc, char *argv[])
     WT_CURSOR *cursor1, *cursor2, *jcursor;
     WT_ITEM k, v;
     WT_SESSION *session;
-    int32_t key, val[2];
-    int i, ret;
+    /* To avoid integer overflow we have to use uint32_t instead of int32_t for key and value. */
+    uint32_t i, key, val[2];
```
Comment (tammybailey) [inline test/csuite/wt2999_join_extractor/main.c:90]: You're still going past the max value for an `uint32_t` in the key and value. They are just wrapping around due to the unsigned.

Comment (AlexeyAnisimovAU) [reply test/csuite/wt2999_join_extractor/main.c:90]: You're right thanks!

**Comment by AlexeyAnisimovAU** [comment]
With this change I've changed content of the database. @tetsuo-cpp please verify that the test logic hasn't changed.

**Review by tetsuo-cpp — APPROVED** [review]
LGTM. I find this test kind of confusing though. I think it's fine since it only seems to be specifying keys 20 and 30 (which your insert phase covers).

File: test/csuite/wt2999_join_extractor/main.c (hunk 1 above)
Comment (tammybailey) [reply test/csuite/wt2999_join_extractor/main.c]: Nope, but ideally we'd not do this intentionally :)

**Review by tammybailey — APPROVED** [review]
lgtm!!

**Comment by AlexeyAnisimovAU** [comment]
Rolled back to the original version and changed int32_t to int64_t. I tried that at the first place but the test was crashing with WT_NOTFOUND. Apparently I just needed to modify data type in custom extractors too.

**Pull request merged**